### PR TITLE
Add compatibility for neovim pr 17462

### PR DESCRIPTION
**Pull request by p00f**
see https://github.com/neovim/neovim/pull/17462

### CODE REVIEW

File: fnl/lightspeed.fnl [hunk 1]
```diff
@@ -351,18 +359,21 @@ character instead."
   (if (or ?target-windows omni?)
       (each [_ win (ipairs (or ?target-windows
                                [(. (vim.fn.getwininfo (vim.fn.win_getid)) 1)]))]
-        (vim.highlight.range win.bufnr hl.ns hl.group.greywash
-                             [(dec win.topline) 0] [(dec win.botline) -1] :v false
-                             hl.priority.greywash))
+        (hl-compat win.bufnr hl.ns hl.group.greywash
+                   [(dec win.topline) 0] [(dec win.botline) -1]
+                   {:regtype :charwise
```
Comment (ggandor) [inline fnl/lightspeed.fnl:364]: Charwise should be `:c` or `:v`, according to `:h setreg()`, shouldn't it? (Not that it matters too much at the moment, as `vim.highlight.range` calls `vim.region`, which [apparently](https://github.com/neovim/neovim/blob/f0ace6d419380907b719edd2ea1d2256e92beb84/src/nvim/lua/vim.lua#L441) only checks for `regtype:byte() == 22` (`^V`)...)

File: fnl/lightspeed.fnl [hunk 2]
```diff
@@ -98,6 +98,14 @@ character instead."
     (if (and ok? (= (type eval-rhs) :string)) eval-rhs rhs)))
 
 
+; For pre-0.7 compatibility
+(fn hl-compat [bufnr ns higroup start finish opts]
```
Comment (ggandor) [inline fnl/lightspeed.fnl:102]: This is a bit obscure, let it be `highlight-range-compat` instead.

File: fnl/lightspeed.fnl (hunk 1 above)
Comment (ggandor) [inline fnl/lightspeed.fnl:364]: Charwise should be `:c` or `:v`, according to `:h setreg()`, shouldn't it? (Not that it matters too much at the moment, as `vim.highlight.range` calls `vim.region`, which [apparently](https://github.com/neovim/neovim/blob/f0ace6d419380907b719edd2ea1d2256e92beb84/src/nvim/lua/vim.lua#L441) only checks for `regtype:byte() == 22` (`^V`), but still...)

**Review by ggandor — CHANGES_REQUESTED** [review]
Thanks for the PR! Could you check the comments?

**Comment by p00f** [comment]
Done

File: fnl/lightspeed.fnl [hunk 3]
```diff
@@ -351,18 +359,21 @@ character instead."
...
             [win-top win-bot] [(dec (vim.fn.line "w0")) (dec (vim.fn.line "w$"))]
             [start finish] (if reverse?
                                [[win-top 0] [curline curcol]]
                                [[curline (inc curcol)] [win-bot -1]])]
         ; Expects 0,0-indexed args; `finish` is exclusive.
-        (vim.highlight.range 0 hl.ns hl.group.greywash
-                             start finish :v false
-                             hl.priority.greywash))))
+        (highlight-range-compat 0 hl.ns hl.group.greywash start finish
+                                {:regtype :charwise}
```
Comment (ggandor) [inline fnl/lightspeed.fnl:374]: Also here...

File: fnl/lightspeed.fnl [hunk 4]
```diff
@@ -372,9 +383,10 @@ character instead."
   "A wrapper around `vim.highlight.range` that handles forced motion
 types properly."
   (let [hl-range (fn [start end end-inclusive?]
-                   (vim.highlight.range 0 hl.ns hl-group
-                                        start end :v end-inclusive?
-                                        hl.priority.label))]
+                   (highlight-range-compat 0 hl.ns hl-group start end
+                                           {:regtype :charwise
```
Comment (ggandor) [inline fnl/lightspeed.fnl:387]: ...and here :)

**Review by ggandor — CHANGES_REQUESTED** [review]
Let's change the regtype in those two other places, and then good to go.

**Comment by p00f** [comment]
done

**Comment by muhmud** [comment]
I'm getting since pulling the latest, have gone back one commit:

image

**Comment by megalithic** [comment]
same! 

> I'm getting since pulling the latest, have gone back one commit:
> 
> image

same. :/

**Pull request merged**